04-data-structures-part1.Rmd

```diff
@@ -2,336 +2,256 @@
 
-```{r, eval=FALSE}
-typeof() # what is its atomic type?
-is.logical() # is it TRUE/FALSE data?
-is.numeric() # is it numeric?
-is.integer() # is it an integer?
-is.complex() # is it complex number data?
-is.character() # is it character data?
-str()  # what is it?
+We can begin exploring our dataset right away, pulling out columns and rows via the following:
+
+```{r}
+cats$weight
+cats$weight[1]
+cats$weight[2]
+cats$weight[1] + cats$weight[2]
 ```
 
-> ## Challenge 1: Data types {.challenge}
->
-> Use your knowledge of how to assign a value to
-> a variable, to create examples of data with the
-> following characteristics:
->
-> 1) Variable name: 'answer', Type: logical
-> 2) Variable name: 'height', Type: numeric
-> 3) Variable name: 'dog_name', Type: character
->
-> For each variable you've created, test that it
-> has the data type you intended. Do you find
-> anything unexpected?
->
+But what about
 
-## Data Structures
+```{r}
+cats$weight[1] + cats$coat[2]
+```
 
-There are five data structures you will commonly encounter in R. These are:
+Understanding what happened here is key to successfully analyzing data in R.
 
-* vector
-* factor
-* list
-* matrix
-* data.frame
+## Data Types
 
-For now, let's focus on vectors in more detail, to discover more about data types.
+If you guessed that the last command will return an error because 2.1 plus black is nonsense, you're right - and you already have some intuition for an important concept in programming called *data types*. We can ask what type of data something is:
 
-## Vectors
+```{r}
+typeof(cats$weight[1])
+```
 
-A vector is the most common and basic data structure in `R` and is pretty much
-the workhorse of R. They are sometimes referred to as atomic vectors, because
-importantly, they can **only contain one data type**. They are the building blocks of
-every other data structure.
+There are 5 main types: doubles, integers, complex, logical and character.
 
-A vector can contain any of the five types we introduced before:
+```
+typeof(3.14)
+typeof(1L)
+typeof(1+1i)
+typeof(TRUE)
+typeof('banana')
+```
 
-* logical (e.g., `TRUE`, `FALSE`)
-* integer (e.g., `2L`, `as.integer(3)`)
-* numeric (real or decimal) (e.g, `2`, `2.0`, `pi`)
-* complex (e.g, `1 + 0i`, `1 + 4i`)
-* character (e.g, `"a"`, `"swc"`)
+Note the `L` suffix to insist that a number is an integer. No matter how complicated our analyses become, all data in R is interpreted as one of these basic data types. This strictness has some really important concequences. Try adding this row to your cat data:
 
-> ## Tip: "Character Vectors" {.callout}
->
-> You will sometimes hear the term "character vector", especially in
-> warning or error messages. This is a somewhat confusing and unfortunate
-> name. Remember that the type "character" really means some text
-> wrapped in quotation symbols.
->
+```{r, eval=FALSE}
+tabby,2.3 or 2.4,TRUE
+```
 
```

Comment:
Done, remaining coy about precisely which editor to use - this is definitely clearer, but I share @aammd's concerns over a round trip through a spreadsheet; I'll leave it to the instructor to lead.